lib/improver/argparser.py

```diff
@@ -28,19 +28,185 @@
+                 specific_arguments=None, **kwargs):
+        """Create an ArgParse instance, which is a subclass of
+        argparse.ArgumentParser and automatically add all of the arguments.
+        (Note: The ArgParse.COMPULSORY_ARGUMENTS are always added.)
+
+        Args:
+            central_arguments (list):
+              A list containing the centralized arguments we require.
+              (Keys of the centralized argument dictionary). By default this is
+              set as ArgParse.DEFAULT_CENTRALIZED_ARGUMENTS.
+            specfic_arguments (list):
+              A list of argument specifications required to add arguments
+              which are not contained within the centralized argument
+              dictionary. The format of these argument specifications should
+              be the same as the values in the centralized argument dictionary.
+              (For more details, see the add_arguments method).
+              Default is None, which does not add additional arguments.
+            kwargs (dictionary):
+              Additional keyword arguments which are passed to the superclass
+              constructor (argparse.ArgumentParser).
+        """
+
+        self._args = None
+
+        # Allow either central_arguments or specific_arguments to be None
+        # (or empty lists)
+        if central_arguments is None:
+            central_arguments = []
+        if specific_arguments is None:
+            specific_arguments = []
+
+        # argspecs of the compulsory arguments (no switch here)
+        compulsory_arguments = ArgParser.COMPULSORY_ARGUMENTS.values()
+
+        # get argspecs of the central arguments
+        central_arguments = self._args_from_selected_central_args(
+                             central_arguments)
+
+        # specific arguments must be passed in with the correct format
+        # (argspecs)- we don't need to do anything special here...
+
+        # create instance of ArgumentParser (pass along kwargs)
+        super(ArgParser, self).__init__(**kwargs)
+
+        cli_arguments = (compulsory_arguments + central_arguments +
+                         specific_arguments)
+
+        # automatically add all of the arguments
+        self.add_arguments(cli_arguments)
+        # Done. Now we can get the arguments with self.args()
+
+    @staticmethod
+    def _args_from_selected_central_args(arg_name_list):
+        """Return a list of argument specifications from the
+        ArgParser.CENTRALIZED_ARGUMENTS, given a list of keys (argument names).
+
+        Args:
+            arg_name_list (list):
+              A list containing the keys required from the centralized argument
+              dictionary.
+
+        Returns:
+            list:
+              The argument specifications which may later be used to add
+              arguments to the parser.
+        """
+        return [ArgParser.CENTRALIZED_ARGUMENTS[arg_name] for arg_name
+                in arg_name_list]
```

Comment:
This doesn't need to be a function - can you just put the one-liner into "init" please.

Reply:
Given that this is likely to only be called from the constructor, I'm happy to do this.